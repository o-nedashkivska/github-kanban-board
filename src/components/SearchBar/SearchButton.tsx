import { useSelector } from "react-redux";
import { Button } from "antd";
import { getDataIsLoading } from "../../store/selectors";

interface SearchButtonProps {
  disabled: boolean;
  onClick: () => void;
}

const SearchButton: React.FC<SearchButtonProps> = ({ disabled, onClick }) => {
  const isLoading = useSelector(getDataIsLoading);

  return (
    <Button size="large" disabled={disabled || isLoading} onClick={onClick}>
      Load Issues
    </Button>
  );
};

export default SearchButton;
