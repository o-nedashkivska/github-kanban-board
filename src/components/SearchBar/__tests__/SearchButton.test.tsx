import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { renderWithProviders } from "../../../test-utils";

import SearchButton from "../SearchButton";

describe("SearchButton", () => {
  const onClick = jest.fn();

  test("displays correct button text", () => {
    renderWithProviders(<SearchButton disabled={false} onClick={onClick} />);

    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();

    expect(screen.getByText("Load Issues")).toBeInTheDocument();
  });

  describe("displays correct button when status is loading", () => {
    const preloadedState = {
      currentRepo: { name: "", stars: 0, status: "loading", statusReason: "" },
      issues: {
        status: { toDo: "initial", inProgress: "loading", done: "loading" },
        limit: 0,
        allIssuesByRepo: {},
      },
    };

    test("button is disabled when status is loading and disabled prop is false", () => {
      renderWithProviders(<SearchButton disabled={false} onClick={onClick} />, {
        preloadedState,
      });

      const button = screen.getByRole("button");

      expect(button).toBeInTheDocument();
      expect(button).toBeDisabled();

      userEvent.click(button);
      expect(onClick).not.toHaveBeenCalled();
    });

    test("button is disabled when status is loading and disabled prop is true", () => {
      renderWithProviders(<SearchButton disabled={true} onClick={onClick} />, {
        preloadedState,
      });

      const button = screen.getByRole("button");

      expect(button).toBeInTheDocument();
      expect(button).toBeDisabled();

      userEvent.click(button);
      expect(onClick).not.toHaveBeenCalled();
    });

    test("matches snapshot", () => {
      const { asFragment } = renderWithProviders(
        <SearchButton disabled={false} onClick={onClick} />,
        {
          preloadedState,
        }
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("displays correct button when status is other than loading", () => {
    test("button is disabled when status is other than loading and disabled prop is true", () => {
      renderWithProviders(<SearchButton disabled={true} onClick={onClick} />);

      const button = screen.getByRole("button");

      expect(button).toBeInTheDocument();
      expect(button).toBeDisabled();

      userEvent.click(button);
      expect(onClick).not.toHaveBeenCalled();
    });

    test("button is enabled when status is other than loading and disabled prop is false", () => {
      renderWithProviders(<SearchButton disabled={false} onClick={onClick} />);

      const button = screen.getByRole("button");

      expect(button).toBeInTheDocument();
      expect(button).toBeEnabled();

      userEvent.click(button);
      expect(onClick).toHaveBeenCalledTimes(1);
    });

    test("matches snapshot", () => {
      const { asFragment } = renderWithProviders(
        <SearchButton disabled={true} onClick={onClick} />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
