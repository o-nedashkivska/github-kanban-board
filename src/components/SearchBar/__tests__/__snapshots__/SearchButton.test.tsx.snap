// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton displays correct button when status is loading matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="ant-btn css-dev-only-do-not-override-1kuana8 ant-btn-default ant-btn-lg"
    disabled=""
    type="button"
  >
    <span>
      Load Issues
    </span>
  </button>
</DocumentFragment>
`;

exports[`SearchButton displays correct button when status is other than loading matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="ant-btn css-dev-only-do-not-override-1kuana8 ant-btn-default ant-btn-lg"
    disabled=""
    type="button"
  >
    <span>
      Load Issues
    </span>
  </button>
</DocumentFragment>
`;
